.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+name: "validate pull request"
+
+on:
+  push:
+    branches: [ main, dev ]
+  pull_request:
+    branches: [ main, dev ]
+    # Add assigned to defaults as a hack to easily trigger code-gen PRs to run validation
+    types: [assigned, opened, synchronize, reopened]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  validate-pull-request:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2.3.4
+    - name: composer validate
+      run: |
+        composer validate
+    - name: composer install
+      run: |
+        composer install
+    - name: run tests
+      run : |
+        vendor/bin/phpunit --exclude-group functional
+  run-static-analysis:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2.3.4
+    - name: composer validate
+      run: |
+        composer validate
+    - name: Composer install
+      run: |
+        composer install
+    - name: Run analysis 
```

Comment:
Thinking we could add this as a step under `validate-pull-request` since they have the same run steps